src/analysis/processing/qgsalgorithmsplitvectorlayer.cpp

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************
+                         qgsalgorithmsplitvectorlayer.cpp
+                         ---------------------
+    begin                : May 2020
+    copyright            : (C) 2020 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmsplitvectorlayer.h"
+#include "qgsvectorfilewriter.h"
+
+///@cond PRIVATE
+
+QString QgsSplitVectorLayerAlgorithm::name() const
+{
+  return QStringLiteral( "splitvectorlayer" );
+}
+
+QString QgsSplitVectorLayerAlgorithm::displayName() const
+{
+  return QObject::tr( "Split vector layer" );
+}
+
+QStringList QgsSplitVectorLayerAlgorithm::tags() const
+{
+  return QObject::tr( "vector,split,field,unique" ).split( ',' );
+}
+
+QString QgsSplitVectorLayerAlgorithm::group() const
+{
+  return QObject::tr( "Vector general" );
+}
+
+QString QgsSplitVectorLayerAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeneral" );
+}
+
+QString QgsSplitVectorLayerAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Splits input vector layer into multiple layers by specified unique ID field."
+                      "Each of the layers created in the output folder contains all features from "
+                      "the input layer with the same value for the specified attribute. The number "
+                      ":of files generated is equal to the number of different values found for the "
+                      "specified attribute." );
+}
+
+QgsSplitVectorLayerAlgorithm *QgsSplitVectorLayerAlgorithm::createInstance() const
+{
+  return new QgsSplitVectorLayerAlgorithm();
+}
+
+void QgsSplitVectorLayerAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ) ) );
+  addParameter( new QgsProcessingParameterField( QStringLiteral( "FIELD" ), QObject::tr( "Unique ID field" ),
+                QVariant(), QStringLiteral( "INPUT" ) ) );
+  addParameter( new QgsProcessingParameterFolderDestination( QStringLiteral( "OUTPUT" ), QObject::tr( "Output directory" ) ) );
+  addOutput( new QgsProcessingOutputMultipleLayers( QStringLiteral( "OUTPUT_LAYERS" ), QObject::tr( "Output layers" ) ) );
+}
```

Comment:
What do you feel about adding an optional advanced string parameter for "output file type"? (Not essential)

Reply:
Done. New parameter with available extensions added, I will turn it into more human-friendly driver names later.